apps/backend/src/libs/hooks/check-user-permissions.hook.ts

```diff
@@ -16,6 +16,16 @@ const checkUserPermissions = (routePermissions: string[]): APIPreHandler => {
 			});
 		}
 
+		const resourceId = (params as { id?: string }).id;
+		const isUsersRoute = url.includes(APIPath.USERS);
+		const isCurrentUserRoute = isUsersRoute && Number(resourceId) === user.id;
+
+		if (isCurrentUserRoute) {
+			done();
+
+			return;
+		}
```

Comment:
This logic shouldn't be in checkUserPermissions hook. This hook is for common permission case and should now nothing about specific routes and their logic. Instead let's use DELETE DELETE /authenticated-user and PUT /authenticated-user in auth controller for current user, and move the logic related to this routes to auth module

Reply:
yes